File: platform/android/MapboxGLAndroidSDK/src/test/java/org/maplibre/android/MapLibreInjector.java
package org.maplibre.android;

import android.content.Context;

import androidx.annotation.NonNull;

import org.maplibre.android.util.TileServerOptions;

import java.lang.reflect.Field;

public class MapLibreInjector {

  private static final String FIELD_INSTANCE = "INSTANCE";

  public static void inject(@NonNull Context context, @NonNull String apiKey,
                            @NonNull TileServerOptions options) {
    MapLibre maplibre = new MapLibre(context, apiKey, options);
    try {
      Field instance = MapLibre.class.getDeclaredField(FIELD_INSTANCE);
      instance.setAccessible(true);
      instance.set(maplibre, maplibre);
    } catch (Exception exception) {
      throw new AssertionError();
    }
  }

  public static void clear() {
    try {
      Field field = MapLibre.class.getDeclaredField(FIELD_INSTANCE);
      field.setAccessible(true);
      field.set(field, null);
    } catch (Exception exception) {
      throw new AssertionError();
    }
  }
}

File: platform/android/MapLibreAndroid/src/test/java/org/maplibre/android/MapLibreInjector.kt
package org.maplibre.android

import android.content.Context
import org.maplibre.android.MapLibre
import org.maplibre.android.util.TileServerOptions

object MapLibreInjector {
    private const val FIELD_INSTANCE = "INSTANCE"
    @JvmStatic
    fun inject(context: Context, apiKey: String,
               options: TileServerOptions) {
        val maplibre = MapLibre(context, apiKey, options)
        try {
            val instance = MapLibre::class.java.getDeclaredField(FIELD_INSTANCE)
            instance.isAccessible = true
            instance[maplibre] = maplibre
        } catch (exception: Exception) {
            throw AssertionError()
        }
    }

    @JvmStatic
    fun clear() {
        try {
            val field = MapLibre::class.java.getDeclaredField(FIELD_INSTANCE)
            field.isAccessible = true
            field[field] = null
        } catch (exception: Exception) {
            throw AssertionError()
        }
    }
}
